include/openspace/interaction/orbitalnavigator.h

```diff
@@ -0,0 +1,237 @@
+#include <openspace/interaction/interpolator.h>
+#include <openspace/interaction/mousestate.h>
+
+#include <openspace/properties/propertyowner.h>
+#include <openspace/properties/scalar/boolproperty.h>
+#include <openspace/properties/scalar/floatproperty.h>
+
+#include <glm/glm.hpp>
+#include <glm/gtx/quaternion.hpp>
+
+namespace openspace {
+
+class SceneGraphNode;
+class Camera;
+class SurfacePositionHandle;
+
+namespace globebrowsing {
+    class RenderableGlobe;
+    class TileIndex;
+    class Geodetic2;
+    class Geodetic3;
+}
+
+namespace interaction {
+
+class OrbitalNavigator : public properties::PropertyOwner 
+{
+public:
+    OrbitalNavigator();
+    ~OrbitalNavigator();
+
+    void updateMouseStatesFromInput(const InputState& inputState, double deltaTime);
+    void updateCameraStateFromMouseStates(Camera& camera, double deltaTime);
+    void setFocusNode(SceneGraphNode* focusNode);
+    void startInterpolateCameraDirection(const Camera& camera);
+
+    bool followingNodeRotation() const;    
+    SceneGraphNode* focusNode() const;
+
+#ifdef OPENSPACE_MODULE_GLOBEBROWSING_ENABLED
+    void goToChunk(Camera& camera,
+                   globebrowsing::TileIndex ti,
+                   glm::vec2 uv,
+                   bool resetCameraDirection);
+    void goToGeodetic2(Camera& camera,
+                       globebrowsing::Geodetic2 geo2,
+                       bool resetCameraDirection);
+    
+    void goToGeodetic3(Camera& camera,
+                       globebrowsing::Geodetic3 geo3,
+                       bool resetCameraDirection);
+#endif
+
+private:
+    struct CameraRotationDecomposition {
+        glm::dquat localRotation;
+        glm::dquat globalRotation;
+    };
+
+    // Properties
+    properties::BoolProperty _rotationalFriction;
+    properties::BoolProperty _horizontalFriction;
+    properties::BoolProperty _verticalFriction;
+    properties::FloatProperty _followFocusNodeRotationDistance;
+    properties::FloatProperty _minimumAllowedDistance;
+    properties::FloatProperty _sensitivity;
+    properties::FloatProperty _motionLag;
+
+    MouseStates _mouseStates;
+
+    SceneGraphNode* _focusNode = nullptr;
+    glm::dvec3 _previousFocusNodePosition;
+    glm::dquat _previousFocusNodeRotation;
+    
+    Interpolator<double> _rotateToFocusNodeInterpolator;
+    Interpolator<double> _followRotationInterpolator;
+
+    /**
+     * Decomposes the cameras rotation in to a global and a local rotation defined by
+     * CameraRotationDecomposition. The global rotation defines the rotation so that the
+     * camera points towards the focus node in the direction opposite to the direction
+     * out from the surface of the object. The local rotation defines the differential
+     * from the global to the current total rotation so that
+     * <code>cameraRotation = globalRotation * localRotation</code>. 
+     */
+    CameraRotationDecomposition decomposeCameraRotation(
+        const glm::dvec3& cameraPosition,
```

Comment:
Nitpicky:  In header files it's preferred to fill up the line first before taking a next line